molpipeline/abstract_pipeline_elements/mol2any/mol2bitvector.py

```diff
@@ -196,14 +197,134 @@ def pretransform_single(
         """
 
 
-class ABCMorganFingerprintPipelineElement(MolToFingerprintPipelineElement, abc.ABC):
+class MolToRDKitGenFPElement(MolToFingerprintPipelineElement, abc.ABC):
+    """Abstract class for PipelineElements using the FingeprintGenerator64."""
+
+    def __init__(
+        self,
+        counted: bool = False,
+        return_as: OutputDatatype = "sparse",
+        name: str = "MolToRDKitGenFin",
+        n_jobs: int = 1,
+        uuid: Optional[str] = None,
+    ):
+        """Initialize abstract class.
+
+        Parameters
+        ----------
+        counted: bool
+            Whether to count the bits or not.
+        return_as: Literal["sparse", "dense", "explicit_bit_vect"]
+            Type of output. When "sparse" the fingerprints will be returned as a scipy.sparse.csr_matrix
+        name: str
+            Name of PipelineElement.
+        n_jobs:
+            Number of jobs.
+        uuid: Optional[str]
+            Unique identifier.
+        """
+        super().__init__(
+            return_as=return_as,
+            name=name,
+            n_jobs=n_jobs,
+            uuid=uuid,
+        )
+        self.counted = counted
+
+    @abc.abstractmethod
+    def _get_fp_generator(self) -> rdFingerprintGenerator.FingeprintGenerator64:
+        """Get fingerprint generator.
+
+        Returns
+        -------
+        rdFingerprintGenerator.FingeprintGenerator64
+            Fingerprint generator.
+        """
+
+    def pretransform_single(
+        self, value: RDKitMol
+    ) -> ExplicitBitVect | npt.NDArray[np.int_] | dict[int, int]:
+        """Transform a single compound to a dictionary.
+
+        Keys denote the feature position, values the count. Here always 1.
+
+        Parameters
+        ----------
+        value: RDKitMol
+            Molecule for which the fingerprint is generated.
+
+        Returns
+        -------
+        dict[int, int]
+            Dictionary with feature-position as key and count as value.
+        """
+        fingerprint_generator = self._get_fp_generator()
+        if self._return_as == "explicit_bit_vect":
+            if self.counted:
+                return fingerprint_generator.GetCountFingerprint(value)
+            return fingerprint_generator.GetFingerprint(value)
+
+        if self.counted:
+            fingerprint = fingerprint_generator.GetCountFingerprintAsNumPy(value)
+        else:
+            fingerprint = fingerprint_generator.GetFingerprintAsNumPy(value)
+
+        if self._return_as == "dense":
+            return fingerprint
+
+        return {pos: count for pos, count in enumerate(fingerprint) if count > 0}
```

Comment:
Because the Path Fingerprint returns a `UIntSparseIntVect` which raises:
```bash
AttributeError: 'UIntSparseIntVect' object has no attribute 'GetOnBits'
```
We could add a type check and then use `GetNonzeroElements` instead.  
However, I think the current implementation is clearer.  

Maybe we should ask in RDKit if they would make the interface consistent?

Reply:
Nope, I was wrong. The UIntSparseIntVect comes from the counted FP...  

How could I even doubt RDKit :D

Implemented it now ✅